tutorials/MLlib.md

```diff
@@ -0,0 +1,268 @@
+transformację liczbową na kolumnie SPECIES.
+
+```python
+from pyspark.sql.functions import col, when
+
+iris = (spark.read.format('csv')
+        .option("header", False)
+        .schema(schemat)
+        .load('/content/drive/MyDrive/iris.data')
+)
+iris = (iris.withColumn("SPECIES",
+                       when(col("SPECIES")=='Iris-virginica', 1)
+                       .when(col("SPECIES")=='Iris-setosa', 2)
+                       .when(col("SPECIES")=='Iris-versicolor', 3))
+       .withColumnRenamed("SPECIES", "labels")
+)
+
+iris.show(5)
+```
+```commandline
++-----+-------+-----+-------+------+
+|P_LEN|P_WIDTH|S_LEN|S_WIDTH|labels|
++-----+-------+-----+-------+------+
+|  5.1|    3.5|  1.4|    0.2|     2|
+|  4.9|    3.0|  1.4|    0.2|     2|
+|  4.7|    3.2|  1.3|    0.2|     2|
+|  4.6|    3.1|  1.5|    0.2|     2|
+|  5.0|    3.6|  1.4|    0.2|     2|
++-----+-------+-----+-------+------+
+only showing top 5 rows
+```
+### Podziału label-features
+Klasa `VectorAssembler` transformuje wiele kolumn zawierających cechy i zamienia je 
+na wektor.
+
+```python
+from pyspark.ml.feature import VectorAssembler
+
+assembler = VectorAssembler(
+    inputCols=['P_LEN', 'P_WIDTH', 'S_LEN', 'S_WIDTH'],
+    outputCol="features"
+    )
+stages = [assembler]
+```
+
+Przy tworzeniu modelu użyjemy klasy `Pipeline` z biblioteki pyspark.ml.
+Definiuje ona sekwencję etapów przetwarzania, przekształcania i modelowania danych. 
+Pipeline składa się z szeregu etapów 'Transformers' i 'Estimators', 
+które są wykonywane w określonej kolejności.
+
+```python
+from pyspark.ml import Pipeline
+
+pipeline = Pipeline(stages = stages)
+
+pipelineModel = pipeline.fit(iris)
+
+iris_training_ready = (
+    pipelineModel
+    .transform(iris)
+    .select('labels', 'features')
+)
+
+iris_training_ready.limit(5).toPandas()
+```
+
+```commandline
+   labels	                                         features
+0	2	[5.099999904632568, 3.5, 1.399999976158142, 0....
+1	2	[4.900000095367432, 3.0, 1.399999976158142, 0....
+2	2	[4.699999809265137, 3.200000047683716, 1.29999...
+3	2	[4.599999904632568, 3.0999999046325684, 1.5, 0...
+4	2	[5.0, 3.5999999046325684, 1.399999976158142, 0...
+```
+### Podział na dane treningowe i testowe
+Do podziału danych wystarczy użyć metody `randomSplit`. 
+Podajemy w niej listę wag, według której podzielony będzie nasz zbiór.
+
+```python
+train, test = iris_training_ready.randomSplit([0.9, 0.1])
+print(train.count())
+print(test.count())
+```
+```commandline
+136
+14
+```
+Widzimy, że w zbiorze treningowym znajduje się 136 obserwacji, a w zbiorze 
+testowym 14.
+
+## Przykładowe modele
+
+### Model regresji logistycznej
+Rozpoczniemy od prostego modelu regresji logistycznej.
+
+```python
+from pyspark.ml.classification import LogisticRegression
+
+model = LogisticRegression(featuresCol = 'features', labelCol = 'labels', maxIter = 10)
+```
+Uczymy model na danych treningowych.
+
+```python
+trained_model = model.fit(train)
```

Comment:
Fajnie by było też pokazać jak działa CV https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.ml.tuning.CrossValidator.html?highlight=crossvalidator. Chociaż to może być "overkill", dla tego projektu.

Reply:
CrossValidator dodany